is_core/rest/paginators.py

```diff
@@ -0,0 +1,203 @@
+from functools import reduce
+from operator import or_
+
+from django.db.models.query import QuerySet
+from django.db.models import Q
+from django.db.models.expressions import OrderBy
+from django.utils.translation import ugettext
+
+from chamber.shortcuts import get_object_or_none
+
+from pyston.exception import RESTException
+from pyston.forms import RESTValidationError
+from pyston.paginator import BaseOffsetPaginator, BasePaginator
+from pyston.utils.compatibility import get_last_parent_pk_field_name
+from pyston.utils.helpers import ModelIterableIteratorHelper
+
+from is_core.config import settings
+
+
+def _get_attr(obj, attr):
+    if '__' in attr:
+        rel_obj, rel_attr = attr.split('__')
+        return _get_attr(getattr(obj, rel_obj), rel_attr)
+    else:
+        return getattr(obj, attr)
+
+
+class OffsetBasedPaginator(BaseOffsetPaginator):
+
+    type = 'offset-based-paginator'
+
+    def _get_next_offset(self):
+        if self.total:
+            return self.offset + self.base if self.base and self.offset + self.base < self.total else None
+        else:
+            if not self.base:
+                return None
+            next_offset = self.offset + self.base
+            return next_offset if self.qs[next_offset:next_offset + 1].exists() else None
+
+    def _get_total(self):
+        if isinstance(self.qs, QuerySet):
+            if self.qs[:settings.REST_PAGINATOR_MAX_TOTAL+1].count() > settings.REST_PAGINATOR_MAX_TOTAL:
+                return None
+            else:
+                return self.qs.count()
+        else:
+            return len(self.qs)
+
+
+class OffsetBasedPaginatorWithoutTotal(OffsetBasedPaginator):
+
+    def _get_total(self):
+        return None
+
+
+class CursorBasedPaginator(BasePaginator):
```

Comment:
some code is get from core cursor base paginator, but there are diferences. Headers, class parent, name of query strings, etc. It is hard to use common code

Reply:
Maybe I can move the paginator to the pyston

Reply:
moved